api/api/serializers/media_serializers.py

```diff
@@ -334,6 +295,149 @@ def needs_db(self) -> bool:
+        required=False,
+        max_length=200,
+    )
+    title = serializers.CharField(
+        label="title",
+        help_text="Search by title only. Cannot be used with `q`.",
+        required=False,
+        max_length=200,
+    )
+
+    def validate_q(self, value):
+        return self._truncate(value)
+
+    def validate_creator(self, value):
+        return self._truncate(value)
+
+    def validate_tags(self, value):
+        return self._truncate(value)
+
+    def validate_title(self, value):
+        return self._truncate(value)
+
+
+@extend_schema_serializer(
+    # Hide unstable and internal fields from documentation.
+    # Also see `field_names` below.
+    exclude_fields=[
+        "unstable__sort_by",
+        "unstable__sort_dir",
+        "unstable__authority",
+        "unstable__authority_boost",
+        "unstable__include_sensitive_results",
+        "internal__index",
+    ],
+)
+class MediaCollectionRequestSerializer(MediaListRequestSerializer):
+    """This serializer parses and validates collection path and query parameters."""
+
+    fields_names = [
+        *MediaListRequestSerializer.fields_names,
+        "source",
+        "creator",
+        "tag",
+    ]
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+
+    tag = serializers.CharField(
+        label="tag",
+        help_text="Get the collection for selected tag.",
+        required=False,
+        max_length=200,
+    )
+    creator = serializers.CharField(
+        label="creator",
+        help_text="Get the collection for selected creator. "
+        "Must be used with `source`.",
+        required=False,
+        max_length=200,
+    )
+    source = serializers.CharField(
+        label="source",
+        help_text="Get the collection for selected source. "
+        "Can be used with `creator` to get the creator "
+        "collection from this source.",
+        required=False,
+        max_length=200,
+    )
+
+    def validate_tag(self, value):
+        return self._truncate(value)
+
+    def validate_creator(self, value):
+        return self._truncate(value)
+
+    def validate_source(self, value):
+        """Check whether source is a valid source."""
+        allowed_sources = list(search_controller.get_sources(self.media_type).keys())
+        source = value.lower()
+        if source not in allowed_sources:
+            raise serializers.ValidationError(
+                f"Invalid source `{value}`. Allowed sources are: {allowed_sources}."
+            )
+        return source
+
+    def validate(self, data):
+        data = super().validate(data)
+        if "creator" in self.initial_data and "source" not in self.initial_data:
+            raise serializers.ValidationError(
+                "Cannot set `source` without `creator`. "
+                "Use both of these or neither of them."
```

Comment:
```suggestion
                "Cannot set `creator` without `source`. "
```

I think that was backwards. `or neither of them` also implies to me that you also can't set source without creator, which isn't accurate.